packages/connect-agreement/src/__test__/models/agreement.test.ts

```diff
@@ -237,4 +210,126 @@ describe('Agreement', () => {
       })
     })
   })
+
+  describe('sign', () => {
+    const VERSION_NUMBER = '1'
+    const SIGNER_ADDRESS = '0x0090aed150056316e37fe6dfa10dc63e79d173b6'
+
+    it('returns a sign intent', async () => {
+      const abi = new ethers.utils.Interface(['function sign(uint256)'])
+      const intent = await agreement.sign(SIGNER_ADDRESS, VERSION_NUMBER)
```

Comment:
One thing that I'm still a bit unsure is about calling the return value `intent`. As I understood the intent is only the action itself we are trying to do. The return value of intent is a `TransactionPath`/`ForwardingPath`.

Reply:
Not sure about that too tbh